### DISCUSSION TITLE
Could not find method disable() for arguments [InvalidPackage] on task ':audioplayers_android:lint' of type com.android.build.gradle.tasks.LintGlobalTask.

### DISCUSSION THREAD

**Discussion by faisalansari0367** [bluefireteam/audioplayers#1649] [General]
### Checklist

- [X] I read the [troubleshooting guide](https://github.com/bluefireteam/audioplayers/blob/main/troubleshooting.md) before raising this issue
- [X] I made sure that the issue I am raising doesn't already exist

### Current bug behaviour

When I run the app I got:
FAILURE: Build failed with an exception.

* Where:
Build file '/Users/mohdfaisal/.pub-cache/hosted/pub.dev/audioplayers_android-4.0.0/android/build.gradle' line: 58

* What went wrong:
A problem occurred evaluating project ':audioplayers_android'.
> Could not find method disable() for arguments [InvalidPackage] on task ':audioplayers_android:lint' of type com.android.build.gradle.tasks.LintGlobalTask.

### Expected behaviour

It should compile the app without any problem or it should tell what is wrong here

### Steps to reproduce

Add audioplayers: 5.0.0 in pubspec.yaml on flutter version 3.10.6

 

### Code sample

Just add version 5.0.0 to pubspec on flutter version 3.10.6

### Affected platforms

Android

### Platform details

Android

### AudioPlayers Version

5.0.0

### Build mode

debug

### Screenshots

<img width="1440" alt="Screenshot 2023-08-06 at 2 09 54 AM" src="https://github.com/bluefireteam/audioplayers/assets/61263149/c77a9fe2-60f0-41e3-9d5a-b48472ae7450">

### Logs

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/mohdfaisal/.pub-cache/hosted/pub.dev/audioplayers_android-4.0.0/android/build.gradle' line: 58

* What went wrong:
A problem occurred evaluating project ':audioplayers_android'.
> Could not find method disable() for arguments [InvalidPackage] on task ':audioplayers_android:lint' of type com.android.build.gradle.tasks.LintGlobalTask.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
Running Gradle task 'assembleDebug'...                           2,231ms
Exception: Gradle task assembleDebug failed with exit code 1

file path: android/gradle/wrapper/gradle-wrapper.properties
gradle-wrapper.properties

distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-7.6.1-all.zip

  Flutter doctor:
  Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.6, on macOS 13.4.1 22F770820d darwin-arm64, locale en-IN)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.2)
[✓] VS Code (version 1.81.0)
[✓] Connected device (3 available)
[✓] Network resources

  Output of: flutter doctor -v

[✓] Flutter (Channel stable, 3.10.6, on macOS 13.4.1 22F770820d darwin-arm64, locale en-IN)
    • Flutter version 3.10.6 on channel stable at /Users/mohdfaisal/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision f468f3366c (3 weeks ago), 2023-07-12 15:19:05 -0700
    • Engine revision cdbeda788a
    • Dart version 3.0.6
    • DevTools version 2.23.1

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
    • Android SDK at /Users/mohdfaisal/Library/Android/sdk
    • Platform android-33, build-tools 33.0.1
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14C18
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.81.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.70.0

[✓] Connected device (3 available)
    • sdk gphone64 arm64 (mobile) • emulator-5554 • android-arm64  • Android 13 (API 33) (emulator)
    • macOS (desktop)             • macos         • darwin-arm64   • macOS 13.4.1 22F770820d darwin-arm64
    • Chrome (web)                • chrome        • web-javascript • Google Chrome 115.0.5790.114

[✓] Network resources
    • All expected network resources are available.

• No issues found!

  

### Working on PR

no way

**Comment by shibbs1** [comment]
same issue

**Comment by Gustl22** [comment]
Can you try with version 5.1.0? It's probably #1593

**Comment by TNelen** [comment]
Hey! I'm having the same problem using version 5.1.0

**Reply by Gustl22** [reply]
@TNelen which build gradle version do you use? did you try upgrading your build gradle version?

**Comment by TNelen** [comment]
To fix this i removed 
```
lint {
        disable 'InvalidPackage'
    }
```
    
From the audioplayers_android-4.0.1 build.gradle file in the .pub-cache

**Comment by faisalansari0367** [comment]
I fixed it by upgrading my gradle version from 4.1.0 to 7.4.1 
just add the following in your android > build.gradle file

<img width="794" alt="Screenshot 2023-08-21 at 3 47 03 PM" src="https://github.com/bluefireteam/audioplayers/assets/61263149/ec660a84-f51a-44c4-b282-a498ac26a73b">

classpath 'com.android.tools.build:gradle:7.4.1'
